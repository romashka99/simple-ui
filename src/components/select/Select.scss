@import '../../variables.scss';

.select {
	appearance: none;
	padding: 0 10px;
	border: $border;
	border-radius: $border-radius-base;
	background-color: $fill-color-blank;
	color: $text-color-regular;
	height: $height;
	line-height: $height;
	outline: none;
	box-sizing: border-box;
	cursor: pointer;

	&::placeholder {
		color: $text-color-regular;
	}

	&::before {
		content: '';
		justify-self: end;
		width: 0.8em;
		height: 0.5em;
		background-color: var(--select-arrow);
		-webkit-clip-path: polygon(100% 0%, 0 0%, 50% 100%);
		clip-path: polygon(100% 0%, 0 0%, 50% 100%);
	}

	&::after {
		content: '';
		justify-self: end;
		width: 0.8em;
		height: 0.5em;
		background-color: var(--select-arrow);
		-webkit-clip-path: polygon(100% 0%, 0 0%, 50% 100%);
		clip-path: polygon(100% 0%, 0 0%, 50% 100%);
	}

	&--small {
		font-size: $font-size-small;
		height: $height-small;
		line-height: $height-small;
		padding: 0 7px;
		border-radius: $border-radius-small;
	}

	&--large {
		font-size: $font-size-large;
		padding: 0 15px;
		height: $height-large;
		line-height: $height-large;
	}
}
