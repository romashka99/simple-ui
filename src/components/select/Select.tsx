import React, { DetailedHTMLProps, SelectHTMLAttributes } from 'react';
import './Select.scss';

interface ISelectProps
	extends DetailedHTMLProps<
			SelectHTMLAttributes<HTMLSelectElement>,
			HTMLSelectElement
		>,
		React.AriaAttributes {}

export default function Select({
	className,
	children,
	...props
}: ISelectProps) {
	const classes = `select${className ? ' ' + className : ''}`;

	return (
		<select {...props} data-testid='select' className={classes}>
			{children}
		</select>
	);
}
