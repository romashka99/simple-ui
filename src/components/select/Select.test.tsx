import { render, screen } from '@testing-library/react';
import Select from './Select';

describe('Test Select', () => {
	describe('default', () => {
		test('without attr', () => {
			render(<Select>Click</Select>);
			const select = screen.getByTestId('select');
			expect(select.classList.length).toBe(3);
			expect(select.classList.contains('select')).toBe(true);
			expect(select.classList.contains('select--default')).toBe(true);
			expect(select.classList.contains('select--medium')).toBe(true);
			expect(select.getAttribute('disabled')).toBeNull();
			expect(select).toMatchSnapshot();
		});
	});

	describe('color', () => {});
});
