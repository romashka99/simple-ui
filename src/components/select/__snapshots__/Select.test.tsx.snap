// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Select default without attr 1`] = `
<select
  class="select"
  data-testid="select"
>
  Click
</select>
`;
